internal/exp/metrics/staleness/staleness.go

```diff
@@ -102,3 +102,33 @@ func (s *Staleness[T]) Evict() (identity.Stream, bool) {
 func (s *Staleness[T]) Clear() {
 	s.items.Clear()
 }
+
+type Tracker struct {
+	pq PriorityQueue
+}
+
+func NewTracker() Tracker {
+	return Tracker{pq: NewPriorityQueue()}
+}
+
+func (stale Tracker) Refresh(ts time.Time, ids ...identity.Stream) {
+	for _, id := range ids {
+		stale.pq.Update(id, ts)
+	}
+}
+
+func (stale Tracker) Collect(max time.Duration) []identity.Stream {
```

Comment:
Minor: Can we rename this to `CollectStale()`

Reply:
in processor.go I named it the field `stale staleness.Tracker`, so using it reads as `p.stale.Collect()`.
Renaming this to `p.stale.CollectStale()` which I very slightly like less, because it stutters.

no strong opinion